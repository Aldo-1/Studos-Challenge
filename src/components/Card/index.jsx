import useStyles from './styles'
import { useTheme } from 'react-jss'
import Clock from 'assets/images/clock.svg'
import Cloud from 'assets/images/cloud.svg'
import BgDarkBlue from 'assets/images/bg-blue_card.svg'
import BgBlue from 'assets/images/bg-blue_light.svg'

const Card = ({ date, hour, title, subject, teacher, questionsCompleted, questions, type }) => {
  const theme = useTheme()
  const classes = useStyles({ theme, type, background: type === 1 ? BgDarkBlue : BgBlue })
  return (
    <div className={classes.content}>
      <div className={classes.headerCard}>
        <div className={classes.hour}>
          <img src={Clock} alt="Um relógio"></img>
          <span>Até {date} {hour}</span>
        </div>
        <img src={Cloud} alt="Uma nuuvem com um check" />
      </div>
      <h1 className={classes.title}>{title}</h1>
      <div className={classes.footerCard}>
        <div>
          <span>{subject}</span>
          <span>{teacher}</span>
        </div>
        <span>{questionsCompleted}/{questions}</span>
      </div>
    </div>
  )
}

export default Card