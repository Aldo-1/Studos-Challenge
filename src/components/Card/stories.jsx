import Card from './index'

export default {
  title: 'Card',
  component: Card
}

export const Default = (args) => <Card {...args} />

Default.args = {
  date: "02/06",
  hour: "8h30",
  title: "Lista 03 - Fisica Moderna - Modulo 1 - Modulo 2 - Aula 4",
  subject: "Matem√°tica",
  teacher: "Ricardo",
  questionsCompleted: "0",
  questions: "100",
  type: 1
}
