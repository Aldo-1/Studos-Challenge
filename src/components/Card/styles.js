import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  content: ({ background, theme, type }) => ({
    background: `url(${background}) no-repeat center center / cover, ${type === 1 ? theme.colors['darkBlue'] : theme.colors['blue']}`,
    '& span': {
      display: 'block',
      fontSize: `1.4rem`,
      color: `${theme.colors.white}`
    },
    borderRadius: theme.border.radius,
    padding: theme.spacings.xsmall
  }),
  headerCard: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  hour: ({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    background: theme.colors.red,
    padding: '0.2rem 0.4rem 0.2rem 0.4rem',
    borderRadius: theme.border.radius_two,

    '& span': {
      marginLeft: '0.8rem'
    }
  }),
  title: ({ theme }) => ({
    fontSize: theme.font.sizes.small,
    color: theme.colors.white,
    lineHeight: '1.9rem',
    maxWidth: '32rem',
    margin: `${theme.spacings.small} 0`,
    fontWeight: theme.font.bold
  }),
  footerCard: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center'
  }
})

export default useStyles;