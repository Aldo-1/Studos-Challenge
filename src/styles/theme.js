const theme = {
  colors: {
    darkBlue: '#2555EF',
    blue: '#339CF7',
    grey: '#DCECFA',
    red: '#E65353',
    texts: '#A7A9AB',
    white: '#FFFFFF',
    brown: '#ed8721',
    green: '#02c4c4',
    silver: '#c4c4c4'
  },
  font: {
    fontFamily:
      '-apple-system , BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Open Sans, Helvetica Neue, sans-serif',
    bold: 700,
    sizes: {
      xsmall: '1.4rem',
      small: '1.6rem',
      medium: '1.8rem',
      large: '2.2rem'
    }
  },
  border: {
    radius: '0.6rem',
    radius_two: '0.2rem',
    radius_check: '3.4rem',
    radius_circle: '50%',
  },
  spacings: {
    xxsmall: '0.4rem',
    xsmall: '1.2rem',
    small: '1.6rem',
    medium: '3.0rem',
    large: '3.6rem',
    xlarge: '4.8rem'
  },
  darkMode: {
    colors: {
      background: '3B3B3B'
    }
  }
}

export default theme
