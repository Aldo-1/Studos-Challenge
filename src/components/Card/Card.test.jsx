import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helper'
import Card from './index'
import BgDarkBlue from 'assets/images/bg-blue_card.svg'
import BgBlue from 'assets/images/bg-blue_light.svg'
import theme from 'styles/theme'

const props = {
  date: "02/06",
  hour: "8h30",
  title: "Lista 03 - Fisica Moderna - Modulo 1 - Modulo 2 - Aula 4",
  subject: "Matemática",
  teacher: "Ricardo",
  questionsCompleted: "0",
  questions: "100",
  type: 1
}

describe('<Card />', () => {
  it('should render the component', () => {
    const { container } = renderWithTheme(<Card {...props} />)

    expect(screen.getByRole('heading', { name: /Lista 03 - Fisica Moderna - Modulo 1 - Modulo 2 - Aula 4/i })).toBeInTheDocument()
    expect(screen.getByText('Até 02/06 8h30')).toBeInTheDocument()
    expect(screen.getByText('Matemática')).toBeInTheDocument()
    expect(screen.getByText('Ricardo')).toBeInTheDocument()
    expect(screen.getByText('0/100')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the bgBlueDark', () => {
    const { container } = renderWithTheme(<Card {...props} type={1} />)
    expect(container.firstChild).toHaveStyle({
      background: `url(${BgDarkBlue}) no-repeat center center / cover, ${theme.colors['darkBlue']}`,
    })
  })

  it('should render the bgBlue', () => {
    const { container } = renderWithTheme(<Card {...props} type={0} />)
    expect(container.firstChild).toHaveStyle({
      background: `url(${BgBlue}) no-repeat center center / cover, ${theme.colors['blue']}`,
    })
  })
})