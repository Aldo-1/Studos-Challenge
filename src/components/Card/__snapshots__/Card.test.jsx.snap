// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the component 1`] = `
<div
  class="content-0-2-1 content-d0-0-2-6"
>
  <div
    class="headerCard-0-2-2"
  >
    <div
      class="hour-0-2-3 hour-d1-0-2-7"
    >
      <img
        alt="Um relógio"
        src="clock.svg"
      />
      <span>
        Até 
        02/06
         
        8h30
      </span>
    </div>
    <img
      alt="Uma nuuvem com um check"
      src="cloud.svg"
    />
  </div>
  <h1
    class="title-0-2-4 title-d2-0-2-8"
  >
    Lista 03 - Fisica Moderna - Modulo 1 - Modulo 2 - Aula 4
  </h1>
  <div
    class="footerCard-0-2-5"
  >
    <div>
      <span>
        Matemática
      </span>
      <span>
        Ricardo
      </span>
    </div>
    <span>
      0
      /
      100
    </span>
  </div>
</div>
`;
